src/coroutine/memoset/mod.rs

```diff
@@ -178,69 +183,222 @@ impl<F: LurkField> CircuitTranscript<F> {
     }
 }
 
+#[derive(Debug)]
+pub struct Provenance {
```

Comment:
I think this struct would greatly benefit from documentation, since it encodes a (new) key concept